test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1

```diff
@@ -0,0 +1,269 @@
+            if (!$isWindows) {
+                $result1.Replies[0].Options.DontFragment | Should -BeNullOrEmpty
+                $result2.Replies[0].Status               | Should -BeExactly "Success"
+            } else {
+                $result1.Replies[0].Options.DontFragment | Should -BeFalse
+                # We expect 'TtlExpired' but if a router don't reply we get `TimeOut`
+                $result2.Replies[0].Status               | Should -BeIn "TtlExpired","TimedOut"
+            }
+        }
+
+        It "Force IPv6" -Pending {
+            $result = Test-Connection $targetName -Count 1 -IPv6
+
+            $result.Replies[0].Address | Should -BeExactly $targetAddressIPv6
+            # We should check Null not Empty!
+            $result.Replies[0].Options | Should -Be $null
+        }
+
+        It "MaxHops Should -Be greater 0" {
+            { Test-Connection $targetName -MaxHops 0 }  | ShouldBeErrorId "System.ArgumentOutOfRangeException,Microsoft.PowerShell.Commands.TestConnectionCommand"
+            { Test-Connection $targetName -MaxHops -1 } | ShouldBeErrorId "ParameterArgumentValidationError,Microsoft.PowerShell.Commands.TestConnectionCommand"
+        }
+
+        It "Count Should -Be greater 0" {
+            { Test-Connection $targetName -Count 0 }  | ShouldBeErrorId "ParameterArgumentValidationError,Microsoft.PowerShell.Commands.TestConnectionCommand"
+            { Test-Connection $targetName -Count -1 } | ShouldBeErrorId "ParameterArgumentValidationError,Microsoft.PowerShell.Commands.TestConnectionCommand"
+        }
+
+        It "Delay Should -Be greater 0" {
+            { Test-Connection $targetName -Delay 0 }  | ShouldBeErrorId "ParameterArgumentValidationError,Microsoft.PowerShell.Commands.TestConnectionCommand"
+            { Test-Connection $targetName -Delay -1 } | ShouldBeErrorId "ParameterArgumentValidationError,Microsoft.PowerShell.Commands.TestConnectionCommand"
+        }
+
+        It "Delay works" {
+            $result1 = measure-command {Test-Connection localhost -Count 2}
+            $result2 = measure-command {Test-Connection localhost -Delay 4 -Count 2}
+
+            $result1.TotalSeconds | Should -BeGreaterThan 1
+            $result1.TotalSeconds | Should -BeLessThan 3
+            $result2.TotalSeconds | Should -BeGreaterThan 4
+        }
+
+        It "BufferSize Should -Be between 0 and 65500" {
+            { Test-Connection $targetName -BufferSize 0 }     | Should Not Throw
+            { Test-Connection $targetName -BufferSize -1 }    | ShouldBeErrorId "ParameterArgumentValidationError,Microsoft.PowerShell.Commands.TestConnectionCommand"
+            { Test-Connection $targetName -BufferSize 65501 } | ShouldBeErrorId "ParameterArgumentValidationError,Microsoft.PowerShell.Commands.TestConnectionCommand"
+        }
+
+        It "BufferSize works" {
+            $result = Test-Connection $targetName -Count 1 -BufferSize 2
+
+            if (!$isWindows) {
```

Comment:
Can you add a comment about why we have the special case?

Reply:
The comment is in Line 44 - should we duplicate it on every `if (!$isWindows) { `?